OpenEdXMobile/src/main/java/org/edx/mobile/module/analytics/SegmentAnalytics.java

```diff
@@ -615,26 +616,26 @@ public void trackNotificationTapped(@Nullable String courseId) {
     }
 
     @Override
-    public void courseDetailShared(String courseId, String aboutUrl, String shareType) {
+    public void courseDetailShared(String courseId, String aboutUrl, ShareUtils.ShareType method) {
         Event aEvent = new Event();
         aEvent.properties.putValue(Keys.NAME, Values.SOCIAL_COURSE_DETAIL_SHARED);
 
         aEvent.data.putValue(Keys.NAME, courseId);
         aEvent.data.putValue(Keys.CATEGORY, Values.SOCIAL_SHARING);
         aEvent.data.putValue(Keys.URL, aboutUrl);
-        aEvent.data.putValue(Keys.TYPE, shareType);
+        aEvent.data.putValue(Keys.TYPE, Analytics.Util.getShareTypeValue(method));
```

Comment:
As an implementer of the `Analytics` interface, this class should be able to resolve it's the inner classes defined in the interface directly (i.e. the `Analytics.` is not needed here). Alternatively, you could statically import all the methods in the utility class, which would then enable these methods to be resolved directly without needing any namespacing (just like it would be if the static method were defined directly in the interface).